core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -569,28 +569,34 @@ private IndexShardState changeState(IndexShardState newState, String reason) {
         return result;
     }
 
+    public Engine.NoOp prepareMarkingSeqNoAsNoOp(long seqNo, String reason) {
+        verifyReplicationTarget();
+        long startTime = System.nanoTime();
+        return new Engine.NoOp(seqNo, primaryTerm, Engine.Operation.Origin.REPLICA, startTime, reason);
+    }
+
+    public Engine.NoOpResult markSeqNoAsNoOp(Engine.NoOp noOp) throws IOException {
```

Comment:
This was my ask. I find `IndexShard.noOp` to be confusing - why are we doing nothing?. I would be good with renaming the engine method to match the one here. @dakrone would you be good with that?